wfs/1.0.0/spatial-wfs-v1_0_0-source/src/main/java/org/codice/ddf/spatial/ogc/wfs/v1_0_0/catalog/source/WfsFilterDelegate.java

```diff
@@ -1325,4 +1351,115 @@ private double metersToDegrees(double distance) {
+        } catch (JAXBException jbe) {
+            LOGGER.error("Unable to create MultiPolygonType with geometry: [{}]", geometry);
+        }
+        return jaxbElement;
+    }
+
+    private JAXBElement<GeometryCollectionType> createGeometryCollection(Geometry geometry) {
+        JAXBElement<GeometryCollectionType> jaxbElement = null;
+        try {
+            String gml = Wfs10JTStoGML200Converter.convertGeometryToGML(geometry);
+            GeometryCollectionType geometryCollectionType = (GeometryCollectionType) Wfs10JTStoGML200Converter
+                    .convertGMLToGeometryType(gml, Wfs10Constants.GEOMETRY_COLLECTION);
+            jaxbElement = (JAXBElement<GeometryCollectionType>) Wfs10JTStoGML200Converter
+                    .convertGeometryTypeToJAXB(geometryCollectionType);
+        } catch (JAXBException jbe) {
+            LOGGER.error("Unable to create MultiPolygonType with geometry: [{}]", geometry);
+        }
+        return jaxbElement;
+    }
+
+    private JAXBElement<? extends AbstractGeometryType> createGeometryOperand(String wkt) {
+        String convertedWkt = wkt;
+        Geometry wktGeometry = null;
+        try {
+            wktGeometry = getGeometryFromWkt(convertedWkt);
+        } catch (ParseException e) {
+            throw new IllegalArgumentException(
+                    "Unable to parse WKT Geometry [" + convertedWkt + "]", e);
+        }
+        if (wktGeometry instanceof Polygon) {
+            if (isGeometryOperandSupported(Wfs10Constants.POLYGON)) {
+                return createPolygon(convertedWkt);
+            }
+        } else if (wktGeometry instanceof Point) {
+            if (isGeometryOperandSupported(Wfs10Constants.POINT)) {
+                return createPoint(convertedWkt);
+            }
+        } else if (wktGeometry instanceof LineString) {
+            if (isGeometryOperandSupported(Wfs10Constants.LINESTRING)) {
+                return createLineString(wktGeometry);
+            }
+        } else if (wktGeometry instanceof MultiPoint) {
+            if (isGeometryOperandSupported(Wfs10Constants.MULTI_POINT)) {
+                return createMultiPoint(wktGeometry);
+            }
+        } else if (wktGeometry instanceof MultiLineString) {
+            if (isGeometryOperandSupported(Wfs10Constants.MULTI_LINESTRING)) {
+                return createMultiLineString(wktGeometry);
+            }
+        } else if (wktGeometry instanceof MultiPolygon) {
+            if (isGeometryOperandSupported(Wfs10Constants.MULTI_POLYGON)) {
+                return createMultiPolygon(wktGeometry);
+            }
+        } else if (wktGeometry instanceof GeometryCollection) {
+            if (isGeometryOperandSupported(Wfs10Constants.GEOMETRY_COLLECTION)) {
+                return createGeometryCollection(wktGeometry);
+            }
+        }
+        throw new IllegalArgumentException("Unable to create Geometry from WKT String");
```

Comment:
Instead of throwing one generic IAE, I would suggest adding some else clauses and give more specific error descriptions.  For example, you could do the Polygon operand is not supported or geometry is not a supported instance.